libconnman-qt/vpnmodel.h

```diff
@@ -34,9 +34,9 @@
 #define VPNMODEL_H
 
 #include <QAbstractListModel>
+#include "vpnmanager.h"
 
 class VpnModelPrivate;
-class VpnManager;
```

Comment:
There is a QML property that uses this type, and therefore it isn't possible anymore in Qt6 to forward-declare the associated type.
There are two ways to fix this:
* as it has been done here, with a simple include
* using the Q_MOC_INCLUDE macro, to include it during moc processing

See: https://doc.qt.io/qt-6/qtcore-changes-qt6.html#type-registration

Reply:
@rainemak Happy to try it with Q_MOC_INCLUDE if that's more to your liking and adjust accordingly

Reply:
@rainemak: Tested with Q_MOC_INCLUDE that seems fine as well.

Reply:
Q_MOC_INCLUDE was introduced in qt 6.0 so a common #include should be cleaner here, avoiding version specific handling.

Reply:
> Q_MOC_INCLUDE was introduced in qt 6.0 so a common #include should be cleaner here, avoiding version specific handling.

I can change it back if both @pvuorela @rainemak are happy with that approach

Reply:
Even if used, it would need to be protected with qt version #ifdef and something else provided for Qt5. I wouldn't expect Raine either favoring such. Just about the first time we are adding Qt6 support so the forward declaration vs #include on Q_PROPERTIES wasn't a self-evident change.

Reply:
@pvuorela @rainemak Made the Q_MOC_INCLUDE only for Qt 6 with https://github.com/sailfishos/libconnman-qt/pull/9/commits/b20683f2646a60406b03788989e605f1c31ee98f That should work in both 5.x and 6.x versions.

Reply:
Should be good enough too. But maybe on next one just do the basic #include.

Reply:
> Should be good enough too. But maybe on next one just do the basic #include.

Which next one?

Reply:
Meaning if you do another similar pull request on another project.